includes/admin/reports/class-wc-report-downloads.php

```diff
@@ -0,0 +1,256 @@
+		parent::__construct( array(
+			'singular'  => 'download',
+			'plural'    => 'downloads',
+			'ajax'      => false,
+		) );
+	}
+
+	/**
+	 * Don't need this.
+	 *
+	 * @param string $position
+	 */
+	public function display_tablenav( $position ) {
+
+		if ( 'top' !== $position ) {
+			parent::display_tablenav( $position );
+		}
+	}
+
+	/**
+	 * Output the report.
+	 */
+	public function output_report() {
+
+		$this->prepare_items();
+
+		// Subtitle for permission if set.
+		if ( isset( $_GET['permission_id'] ) && is_numeric( $_GET['permission_id'] ) && $_GET['permission_id'] > 0 ) {
+			$permission_id = $_GET['permission_id'];
+			// Load the permission, order, etc. so we can render more information.
+			$permission	= null;
+			$product	= null;
+			try {
+				$permission	= new WC_Customer_Download( $permission_id );
+				$product	= wc_get_product( $permission->product_id );
+			} catch ( Exception $e ) {
+				wp_die( sprintf( __( 'Permission #%d not found.', 'woocommerce' ), $permission_id ) );
+			}
+
+			if ( ! empty( $permission ) && ! empty( $product ) ) {
+				// File information.
+				$file = $product->get_file( $permission->get_download_id() );
+				
+				// Output the titles at the top of the report.
+				$this->output_report_title( $file, $product, $permission );
+			}
+		}
+
+		echo '<div id="poststuff" class="woocommerce-reports-wide">';
+		$this->display();
+		echo '</div>';
+	}
+
+	/**
+	 * Output the title at the top of the report.
+	 *
+	 * @param mixed $file
+	 * @param WC_Product $product
+	 * @param WC_Customer_Download $permission
+	 */
+	protected function output_report_title( $file, $product, $permission ) {
+		// Output file information (if provided).
+		if ( $file ) {
+			echo '<p><strong>' . __( 'File:', 'woocommerce' ) . '</strong><br/>' . esc_html( $file->get_name() ) . '</p>';
+		}
+
+		// Output product information.
+		echo '<p><strong>' . __( 'Product: ', 'woocommerce' ) . '</strong><br/>';
+		edit_post_link( $product->get_formatted_name(), '', '', $product->get_id() );
+		echo '</p>';
+
+		// Output order information.
+		echo '<p><strong>' . __( 'Order: ', 'woocommerce' ) . '</strong><br/>';
+		edit_post_link( '#' . $permission->order_id, '', '', $permission->order_id );
+		echo '</p>';
+	}
+
+	/**
+	 * Get column value.
+	 *
+	 * @param mixed $item
+	 * @param string $column_name
+	 */
+	public function column_default( $item, $column_name ) {
+
+		$permission = null;
+		$product = null;
+		try {
+			$permission	= new WC_Customer_Download( $item->permission_id );
+			$product	= wc_get_product( $permission->product_id );
+		} catch ( Exception $e ) {
+			// Ok to continue rendering other information even if permission and/or product is not found.
+		}
+
+		switch ( $column_name ) {
+			case 'timestamp' :
+				echo esc_html( $item->timestamp );
+				break;
+
+			case 'product' :
+				if ( empty( $product ) ) {
+					break;
+				}
+
+				edit_post_link( $product->get_formatted_name(), '', '', $product->get_id() );
```

Comment:
Minor: for consistency with `order`, can we move this inside the `if` and reverse the condition?